digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala

```diff
@@ -288,10 +288,12 @@ Returns empty result as Json message, not as page not found
     }
     val (optionalLon, optionalLat, optionalLinkId, bearing) = massTransitStopPositionParameters(parsedBody)
     val properties = (parsedBody \ "properties").extractOpt[Seq[SimpleProperty]].getOrElse(Seq())
-    val linkId = (parsedBody \ "linkId").extractOpt[Long]
+    val id = params("id").toLong
+    val persistedAsset = massTransitStopService.getPersistedAssetsByIds(Set(id)).headOption
+    val roadLinks = persistedAsset.flatMap { asset => roadLinkService.getRoadLinkAndComplementaryFromVVH(asset.linkId) }

```

Comment:
This val roadLinks =  persistedAsset. ... coulde be removed and send to validateBusStopDirections(properties, asset.linkId)